.grit/patterns/hathora_to_new_ts_sdk.md

```diff
@@ -0,0 +1,363 @@
+# Upgrade Hathora to Dedicated TS SDK
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern rewrite_method_call() {
+  or {
+    `await $x.createApp($args)` => `(await $x.createApp($args)).application`,
+    `$x.createBuild($args)` where or {
+      $args <: [$a, $b] => `$a`
+    } => `$x.createBuild($args)`,
```

Comment:
A rewrite where you do nothing is not necessary / useful.

Reply:
This is rewriting the args by dropping `$b`

Reply:
Yes, and you already did that in the inner rewrite. There's no need to make a superfluous rewrite at the top level.

```suggestion
    `$x.createBuild($args)` where or {
      $args <: [$a, $b] => `$a`
    }
```

Reply:
I see; I misunderstood. Gotcha 👍